darts/models/filtering_model.py

```diff
@@ -0,0 +1,108 @@
+"""
+Filtering Model Base Class
+------------------------------
+
+A filtering model captures the measured values of a time series as a function of the past as follows:
+
+.. math:: y_{t+1} = f(y_{t+1}, y_t, y_{t-1}, ..., y_1),
+
+where :math:`y_t` represents the time series' filtered value(s) at time :math:`t`.
+
+The main functions are `fit()` and `predict()`. `fit()` learns the function `f()`, over the history of
+one or several time series. The function `predict()` applies `f()` on one or several time series in order
+to obtain more accurate predictions of the measured values(s)
+"""
+
+from typing import Optional
+from itertools import product
+from abc import ABC, abstractmethod
+from inspect import signature
+import numpy as np
+import pandas as pd
+
+from ..timeseries import TimeSeries
+from ..logging import get_logger, raise_log, raise_if_not
+
+logger = get_logger(__name__)
+
+
+class FilteringModel(ABC):
```

Comment:
I'm not sure `FilteringModel` is the best name. To me filtering time series really sounds too much like removing low frequency or high frequency components. How about something like `SmoothingModel` ?

Reply:
agree. Renamed in the next iteration